lib/defopt.py

```diff
@@ -1057,7 +1057,7 @@ def _is_constructible_from_str(type_, *, skip_first_arg=False):
     except ValueError:
         # Can be raised for classes, if the relevant info is in `__init__`.
         if not isinstance(type_, type):
-            raise
+            return False
```

Comment:
It may(?) be clearer to write
```python
try:
    ...
except ValueError:
    # No relevant info in signature.  Continue below, to check type_.__init__ if type_ is indeed a type.
    pass
```

Reply:
Since I'm not overly familiar with the code any more, I've decided to just fix the error and leave the surrounding code alone. You're more than welcome to refactor if you like.